build.cake

```diff
@@ -104,39 +104,96 @@ Task("Build")
-               .WithProperty("TargetFramework", framework)
-               .WithProperty("PublishDir", BIN_DIR + framework));
+    foreach (var framework in new[] { "netstandard2.0" })
+        MSBuild(ENGINE_API_PROJECT, CreateMSBuildSettings("Publish")
+            .WithProperty("TargetFramework", framework)
+            .WithProperty("PublishDir", BIN_DIR + framework));
 
-        foreach (var framework in new [] { "netcoreapp2.1", "netcoreapp3.1" })
-             MSBuild(ENGINE_TESTS_CSPROJ, CreateMSBuildSettings("Publish")
-                .WithProperty("TargetFramework", framework)
-                .WithProperty("PublishDir", BIN_DIR + framework));
+    foreach (var framework in new[] { "netcoreapp2.1", "netcoreapp3.1", "net5.0" })
+        MSBuild(ENGINE_TESTS_PROJECT, CreateMSBuildSettings("Publish")
+            .WithProperty("TargetFramework", framework)
+            .WithProperty("PublishDir", BIN_DIR + framework));
 
-        MSBuild(CONSOLE_CSPROJ, CreateMSBuildSettings("Publish")
-            .WithProperty("TargetFramework", "netcoreapp3.1")
-            .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
+    MSBuild(CONSOLE_PROJECT, CreateMSBuildSettings("Publish")
+        .WithProperty("TargetFramework", "netcoreapp3.1")
+        .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
 
-         MSBuild(CONSOLE_TESTS_CSPROJ, CreateMSBuildSettings("Publish")
-            .WithProperty("TargetFramework", "netcoreapp3.1")
-            .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
+    MSBuild(CONSOLE_TESTS_PROJECT, CreateMSBuildSettings("Publish")
+        .WithProperty("TargetFramework", "netcoreapp3.1")
+        .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
+}
 
-    });
+private void BuildEachProjectSeparatelyOnLinux()
+{
+    DotNetRestore(SOLUTION_FILE);
+
+    BuildProject(ENGINE_PROJECT, "net20", "netstandard2.0", "netcoreapp3.1");
+    BuildProject(CONSOLE_PROJECT, "net20", "netcoreapp3.1");
+    BuildProject(AGENT_PROJECT, "net20", "net40", "netcoreapp3.1", "net5.0");
+    BuildProject(AGENT_X86_PROJECT, "net20", "net40");
+
+    BuildProject(ENGINE_TESTS_PROJECT, "net35", "netcoreapp2.1", "netcoreapp3.1", "net5.0");
+    BuildProject(CONSOLE_TESTS_PROJECT, "net35", "netcoreapp3.1");
+
+    BuildProject(MOCK_ASSEMBLY_X86_PROJECT, "net35", "net40", "netcoreapp2.1", "netcoreapp3.1");
+    BuildProject(NOTEST_PROJECT, "net35", "netcoreapp2.1", "netcoreapp3.1");
+
+
+    MSBuild(ENGINE_PROJECT, CreateMSBuildSettings("Publish")
+       .WithProperty("TargetFramework", "netstandard2.0")
+       .WithProperty("PublishDir", BIN_DIR + "netstandard2.0"));
+    CopyFileToDirectory(
+       BIN_DIR + "netstandard2.0/testcentric.engine.metadata.dll",
+       BIN_DIR + "netcoreapp2.1");
+    MSBuild(ENGINE_TESTS_PROJECT, CreateMSBuildSettings("Publish")
+       .WithProperty("TargetFramework", "netcoreapp2.1")
+       .WithProperty("PublishDir", BIN_DIR + "netcoreapp2.1"));
```

Comment:
How come these are only done for some of the targets?

E.g. in `BuildUsingMSBuildOnWindowsOrMacOS` we build both `"netstandard2.0"` and `"netcoreapp3.1"` for `ENGINE_PROJECT`, but in `BuildUsingMSBuildOnWindowsOrMacOS` we only build `"netstandard2.0"` ?

Also we don't build `ENGINE_API_PROJECT` on Linux ?

Reply:
In the new (i.e. linux) code, I actually experimented to determine what was copied correctly without publishing. I actually ascertained that 3.1 and 5.0 didn't need it at all. In the case of netcoreapp2.1, Publish failed to copy the the metadata assembly reference so I resorted to copying it from netstandard2.0.

ENGINE_API_PROJECT is built implicitly when we build ENGINE_PROJECT.

I strongly suspect that we are doing more than we need to with Publish in the Windows build, but I preferred not to mess with it right now.